src/pyasm/command/file_upload.py

```diff
@@ -175,12 +180,38 @@ def _dump_file_to_temp(my):
             return
 
         System().makedirs(dirname)
+   
+        
+        # Determine if base_decode is necessary
+        base_decode = my.base_decode
+        if my.write_mode == "ab":
+            # Check for base_decode indicator file
+            if (os.path.exists("%s.action" % tmp_file_path)):
+                base_decode = True
+        elif base_decode:
+            # Create indicator file if base_decode is necessary
+            decode_action_path = "%s.action" % tmp_file_path
+            f_action = open(decode_action_path, 'w')
+            f_action.write("base64decode")
+            f_action.close()
+
+        # Get temporary file path to read from
+        # Linux uses mkstemp, while Windows uses TemporaryFile
+        if os.name == 'nt':
+            data = my.field_storage.file
+        else:
+            path = my.field_storage.get_path()
+            data = open(path, 'rb')
 
-        data = my.field_storage.file
-
-        # write file to tmp directory
+        # Write file to tmp directory
         f = open("%s" % tmp_file_path, my.write_mode)
+       
+        # Use base 64 decode if necessary.
+        if base_decode:
+            import base64
```

Comment:
what if it is already in ab mode, why are we always reading the 22 bytes ?
wrong place to put import base64